gradio/blocks.py

```diff
@@ -997,6 +1034,36 @@ def preprocess_data(self, fn_index: int, inputs: List[Any], state: Dict[int, Any
             processed_input = inputs
         return processed_input
 
+    def validate_outputs(self, fn_index: int, predictions: List[Any]):
+        block_fn = self.fns[fn_index]
+        dependency = self.dependencies[fn_index]
+
+        dep_outputs = dependency["outputs"]
+
+        if len(predictions) < len(dep_outputs):
```

Comment:
In the case of a single output, predictions will not be a list and `len` would fail, e.g. for a `Number`

Reply:
I'm talking about this case

```python
import gradio as gr

with gr.Blocks() as demo:

    num1 = gr.Number()
    num2 = gr.Number()
    
    btn = gr.Button(value='1')
    btn_2 = gr.Button(value='2')

    btn.click(lambda x: x, num1, num2,)
    btn_2.click(lambda a: a, num1, [num1, num2])

demo.launch()
```

I get `TypeError: object of type 'float' has no len()` when clicking button 2

Reply:
Did you get a chance to look into this @space-nuko? Happy to help if needed and get this PR across the finish line

Reply:
Sorry got busy, should be resolved now